MatrixSDK/Background/MXBackgroundStore.swift

```diff
@@ -24,57 +24,32 @@ enum MXBackgroundStoreErrorCode: Int {
     case userIDMissing = 1001   // User ID is missing in credentials
 }
 
-/// Fake memory store implementation. Uses some real values from an MXFileStore instance.
-class MXBackgroundStore: MXMemoryStore {
+/// Minimalist MXStore implementation. It uses some real values from an MXFileStore instance.
+class MXBackgroundStore: NSObject, MXStore {
 
-    //  In-memory value for eventStreamToken. Will be used as eventStreamToken if provided.
-    private var lastStoredEventStreamToken: String?
-    private var credentials: MXCredentials
     //  real store
     private var fileStore: MXFileStore
-    private var myUser: MXUser?
+    
+    // Room stores cache
+    private var roomsStore: [String: MXFileRoomStore] = [:]
     
     init(withCredentials credentials: MXCredentials) {
-        self.credentials = credentials
         fileStore = MXFileStore(credentials: credentials)
         //  load real eventStreamToken
         fileStore.loadMetaData()
     }
     
-    override func open(with credentials: MXCredentials, onComplete: (() -> Void)?, failure: ((Error?) -> Void)? = nil) {
-        super.open(with: credentials, onComplete: {
-            guard let userId = credentials.userId else {
-                failure?(NSError(domain: MXBackgroundStoreErrorDomain,
-                                 code: MXBackgroundStoreErrorCode.userIDMissing.rawValue,
-                                 userInfo: nil))
-                return
-            }
-            //  load session user before calling onComplete
-            self.fileStore.asyncUsers(withUserIds: [userId], success: { (users) in
-                if let user = users.first {
-                    self.myUser = user
-                }
-                onComplete?()
-            }, failure: failure)
```

Comment:
We need this, commented about at `user(withUserId userId: String) -> MXUser?` method.

Reply:
I will check again why we need `user(withUserId userId: String) -> MXUser?`
My issue was this is an async call within a synchronous method. I saw cases where the code was doing things whereas the async call was not complete.